### PR TITLE
community(you): Better support for You.com News API

### PR DESCRIPTION
**Pull request by christeefy**
## Description
While `YouRetriever` supports both You.com's Search and News APIs, news is supported as an afterthought. 
More specifically, not all of the News API parameters are exposed for the user, only those that happen to overlap with the Search API. 

This PR:
- improves support for both APIs, exposing the remaining News API parameters while retaining backward compatibility
- refactor some REST parameter generation logic
- updates the docstring of `YouSearchAPIWrapper`
- add input validation and warnings to ensure parameters are properly set by user
- 🚨 Breaking: Limit the news results to `k` items

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

File: libs/community/langchain_community/utilities/you.py [hunk 1]
```diff
@@ -94,6 +113,74 @@ def validate_environment(cls, values: Dict) -> Dict:
...
+            "safesearch": self.safesearch,
+            "country": self.country,
+            **kwargs,
+        }
+
+        # Add endpoint-specific params
+        if self.endpoint_type in ("search", "snippet"):
+            params.update(
+                query=query,
+                num_web_results=self.num_web_results,
```
Comment (ccurme) [inline libs/community/langchain_community/utilities/you.py:170]: double-checking that `num_web_results` is not relevant to the news API, as previously that was passed in?

**Review by ccurme — APPROVED** [review]
Looks good, thanks for flagging the breaking change. IMO I'd consider it a bug that we did not filter to top-k before 🙂

File: libs/community/langchain_community/utilities/you.py (hunk 1 above)
Comment (christeefy) [reply libs/community/langchain_community/utilities/you.py:170]: The [News API](https://documentation.you.com/api-reference/news) has a `count` parameter in lieu of `num_web_results`. Previously `num_web_results` was passed in but the API didn't complain.

**Comment by christeefy** [comment]
Thanks for the review @ccurme. I'll fix the failing CI, and re-submit. 

I also am planning to alias `endpoint_type` with `endpoint`, just to keep it standardized with the You implementations in other frameworks.

**Comment by baskaryan** [comment]
> Thanks for the review @ccurme. I'll fix the failing CI, and re-submit.
> 
> I also am planning to alias `endpoint_type` with `endpoint`, just to keep it standardized with the You implementations in other frameworks.

to run formatter and linter locally
```bash
cd libs/community
poetry install --with lint,typing
make format
make lint
```

**Comment by christeefy** [comment]
From debugging the failing tests, I realized that `pydantic_v1.Field`'s `alias` parameter doesn't behave as I thought it should. The problem I faced is described in https://github.com/pydantic/pydantic/issues/3446.

So I'm reverting the use of `alias`es, which implies:
- Reverting back to `endpoint_type` instead of `endpoint` to prevent breaking changes (I'd like some advice on how/when can I implement it for standardization with other libraries. I can also drop deprecations at that time)
- Removing the `count` alias to `num_web_results`, and updated the documentation stating that `num_web_results` will be mapped to `count` for the News API

**Comment by christeefy** [comment]
Kindly following up on this PR, @ccurme. Tqvm!

**Comment by baskaryan** [comment]
> Kindly following up on this PR, @ccurme. Tqvm!

could we resolve the lint issues? can check them locally by running
```bash
cd libs/community
poetry install --with lint, typing
make format
make lint
```

**Comment by christeefy** [comment]
Apologies for the continued failures on CI. I've double checked and re-run them. No more failures on `make lint` for `you.py` (although failures appear for other files irrelevant to this PR). Can you please re-run the CI again? Thank you!

**Comment by christeefy** [comment]
The CI has been passing for [08e8658](https://github.com/langchain-ai/langchain/pull/22622/commits/08e8658f37dcc51e6860b7084f9e167b1d0e9ece) so I'm following up on this PR @baskaryan. Thank in advance!

**Pull request merged**